packages/cydran/src/log/Logger.ts

```diff
@@ -1,143 +1,101 @@
-interface Logger {
-
-	/**
-	 * Get the structured key name of the logger.
-	 * @returns string
-	 */
-	getKey(): string;
-
-	/**
-	 * Get human readable label for the logger. 
-	 * @returns string
-	 */
-	getLabel(): string;
+import { IsLevelType } from 'log/strategy/IsLevelType';
+
+interface Logger extends IsLevelType {
+  /**
+   * Get the structured key name of the logger.
+   * @returns string
+   */
+  getKey(): string;
+
+  /**
+   * Get human readable label for the logger.
+   * @returns string
+   */
+  getLabel(): string;
+
+  /**
+   * Log payload at a "trace" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  trace(primaryMsg: string, ...moreArgs: any[]): void;
+
+  /**
+   * Log payload at a "debug" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  debug(primaryMsg: string, ...moreArgs: any[]): void;
+
+  /**
+   * Log payload at a "info" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  info(primaryMsg: string, ...moreArgs: any[]): void;
+
+  /**
+   * Log payload at a "warn" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  warn(primaryMsg: string, ...moreArgs: any[]): void;
+
+  /**
+   * Log payload at a "error" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  error(primaryMsg: string, ...moreArgs: any[]): void;
+
+  /**
+   * Log payload at a "fatal" level
+   * @param primaryMsg to be written out
+   * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
+   */
+  fatal(primaryMsg: string, ...moreArgs: any[]): void;
 
 	/**
 	 * Only if log at a "trace" level execute fn() to derive payload to log
-	 * @param payloadFn to be written out
-	 * @param error optional if there is an error object with detail data
+	 * @param primaryMsgFn result of to be written out
+	 * @param moreArgs additional optional arguments; the last argument as an Error is assumed to be required for the stack ouptut regardless of log level
 	 */
-	ifTrace(payloadFn: () => any, error?: Error): void;
+	ifTrace(payloadFn: () => any, ...moreArgs: any[]): void;
```

Comment:
It looks like the rename to primaryMsgFn was overlooked on this method definition and some that follow it.

Reply:
fixed